test/ts/Component.tsx

```diff
@@ -1,29 +1,70 @@
-import { h, Component } from "hyperapp"
+import {
+  h,
+  app,
+  LazyComponent,
+  Component,
+  View,
+  ActionType,
+  ActionsType
+} from "hyperapp"
 
 interface State {
   count: number
 }
 
+const state: State = { count: 10 }
+
 interface Actions {
-  down(): State
-  up(value: number): State
+  down: ActionType<State, Actions>
+  up: ActionType<State, Actions>
+}
+
+const actions: ActionsType<State, Actions> = {
+  down: () => state => ({ count: state.count - 1 }),
+  up: (value: number) => state => ({
+    count: state.count + value
+  })
 }
 
-const Counter: Component<{ state: State; actions: Actions }> = ({
-  state,
-  actions
-}) => (
+interface ComponentParams {
+  count: number
+  up: () => void
+  down: () => void
+}
+const Counter: Component<ComponentParams> = ({ count, up, down }) => (
   <div>
-    <div>{state.count}</div>
-    <button onclick={actions.down}>-</button>
-    <button onclick={actions.up}>+</button>
+    <div>{count}</div>
+    <button onclick={down}>-</button>
+    <button onclick={up}>+</button>
   </div>
 )
 
-const LazyCounter: Component<{}, State, Actions> = () => (state, actions) => (
+interface LazyComponentParams {
+  label: string
+}
+const LazyCounter: LazyComponent<LazyComponentParams, State, Actions> = ({
```

Comment:
As I mentioned, `LazyComponent` (or `ComponentCreator`) should be a userland typing. Or better, define it in your own project.

Reply:
Explained above - it seems to be a framework concept to me